packages/ethers-5/src/client.ts

```diff
@@ -0,0 +1,797 @@
+        }
+
+        if (ethersConfig.injected) {
+          if (!window?.ethereum) {
+            return false
+          }
+        }
+
+        return ids.some(id => Boolean(window.ethereum?.[String(id)]))
+      },
+
+      disconnect: async () => {
+        const provider = ProviderController.state.provider
+        const providerType = ProviderController.state.providerType
+        if (providerType === ConstantsUtil.WALLET_CONNECT_CONNECTOR_ID) {
+          const WalletConnectProvider = provider?.provider as EthereumProvider
+          await WalletConnectProvider.disconnect()
+          localStorage.removeItem(WALLET_ID)
+          ProviderController.reset()
+        } else {
+          localStorage.removeItem(WALLET_ID)
+          ProviderController.reset()
+        }
+      }
+    }
+
+    super({
+      networkControllerClient,
+      connectionControllerClient,
+      defaultChain: getCaipDefaultChain(defaultChain),
+      tokens: HelpersUtil.getCaipTokens(tokens),
+      _sdkVersion: _sdkVersion ?? `html-ethers-5-${ConstantsUtil.VERSION}`,
+      ...w3mOptions
+    })
+
+    this.projectId = w3mOptions.projectId
+    this.chains = chains
+
+    this.createProvider()
+
+    ProviderController.subscribeKey('address', () => {
+      this.syncAccount()
+    })
+
+    ProviderController.subscribeKey('chainId', () => {
+      this.syncNetwork(chainImages)
+    })
+
+    this.syncRequestedNetworks(chains, chainImages)
+    this.syncConnectors(ethersConfig)
+
+    if (enableEIP6963) {
+      this.listenConnectors(enableEIP6963)
+      this.privateCheckActive6963Provider()
+    }
+
+    if (ethersConfig.injected) {
+      this.watchInjected(ethersConfig)
+    }
+    if (ethersConfig.coinbase) {
+      this.watchCoinbase(ethersConfig)
+    }
+  }
+
+  // -- Public ------------------------------------------------------------------
+
+  // @ts-expect-error: Overriden state type is correct
+  public override getState() {
+    const state = super.getState()
+
+    return {
+      ...state,
+      selectedNetworkId: HelpersUtil.caipNetworkIdToNumber(state.selectedNetworkId)
+    }
+  }
+
+  // @ts-expect-error: Overriden state type is correct
+  public override subscribeState(callback: (state: Web3ModalState) => void) {
+    return super.subscribeState(state =>
+      callback({
+        ...state,
+        selectedNetworkId: HelpersUtil.caipNetworkIdToNumber(state.selectedNetworkId)
+      })
+    )
+  }
+
+  // -- Private -----------------------------------------------------------------
+  private createProvider() {
+    if (!this.walletConnectProviderInitPromise && typeof window !== 'undefined') {
+      this.walletConnectProviderInitPromise = this.initWalletConnectProvider()
+    }
+
+    return this.walletConnectProviderInitPromise
+  }
+
+  private async initWalletConnectProvider() {
+    const walletConnectProviderOptions: EthereumProviderOptions = {
+      projectId: this.projectId,
+      showQrModal: false,
+      chains: [0],
```

Comment:
```suggestion

```

Let's remove required chains completely, some wallets crash with this config. (tested with Trust & MM)